enhancements/agent-installer/agent-based-installer.md

```diff
@@ -0,0 +1,279 @@
+- Install minor versions of OpenShift that are older than the agent-based
+  installer version.
+- Operate in disconnected environments without the need to separately mirror
+  the release payload to a registry that is usable from the installed cluster.
+
+## Proposal
+
+A command-line tool will enable users to build a custom RHCOS image, initially
+in ISO format, containing the components needed to install. The base ISO will
+be [retrieved](https://github.com/openshift/machine-os-images/) from the
+release payload. Users can then boot this image on the hosts that are to become
+part of the cluster using any method they choose - this could be existing
+automation that they have or manually booting using physical media. One host
+(hereafter referred to as "Node 0") will be selected to run the
+assisted-service. All hosts will run the assisted agent. A single ISO will be
+used for all hosts in the cluster. Required container images will be pulled
+from the payload of the OpenShift release that is being installed.
+
+In the automated workflow, all configuration data for the cluster to be
+installed (including the expected number of hosts) will be provided at the time
+the ISO is generated. Installation will proceed automatically once the
+requisite hosts are available and have passed validations. An API running on
+Node 0 will enable automated tools to monitor progress, detect errors, and
+retrieve logs until the OpenShift control plane is available.
+
+In the interactive flow, the ISO will be generated with only enough data to
+configure networking so that the hosts will be able to communicate with each
+other once they are booted. Users will interact with assisted-service on Node 0
+to define and install the cluster. Unlike the hosted service, each instance of
+assisted-service will work with only a single cluster.
+
+### User Stories
+
+- As a user in an IPv6-only environment with no existing management cluster, I
+  want to deploy a cluster using my own automation for provisioning.
+- As a user in a disconnected environment with no existing management cluster,
+  I want to deploy a cluster using my own automation for provisioning.
+- As a user in a disconnected environment with no existing management cluster,
+  I want to interactively deploy a cluster and receive immediate feedback about
+  configuration errors and problems.
+- As a user in a disconnected environment with no existing management cluster,
+  I want to deploy a cluster to a vSphere where I have permissions only to set
+  the boot image for one or more existing VMs.
+- As a user in a disconnected environment with no existing management cluster
+  and strict security protocols, I  want to deploy a cluster using only
+  physical media and access to the host consoles via a KVM switch.
+
+### API Extensions
+
+N/A
+
+### Implementation Details/Notes/Constraints [optional]
+
+Further details will be discussed in future enhancements.
```

Comment:
This section would be a good place to explain the reasoning behind each of the decisions described in the Proposal section.

"We must be able to run without a service on a host outside of the cluster, therefore we will build images that are self-orchestrating."

"We must support user-provided provisioning tools, therefore we will produce artifacts like ISOs that are ready to be consumed by those tools with minimal integration."

"We must support user-provided manifests like the openshift-installer does and will not have a REST API service, therefore the inputs to the image generator tool will be files."

"We want to minimize the storage requirements for provisioning images, therefore we will build one image that can be used on all hosts instead of creating host-specific images."